### PR TITLE
Improve build documentation

### PR DESCRIPTION
**Pull request by deadeyegoodwin**

### CODE REVIEW

File: docs/build.md [hunk 1]
```diff
@@ -69,62 +104,147 @@ build. For example, if you have a branch called "mybranch" in the
...
+  to find the corresponding packages on libraries on your platform.
+
+* Your platform may use a different compiler or compiler version than
+  the support platforms. As a result you may encounter build errors
+  that need to be fixed by editing the source code or changing the
+  compilation flags.
+
+* Triton depends on a large number of open-source packages that it
+  builds from source. If one of these packages does not support your
+  platform them you may need to diable the Triton feature that depends
```
Comment (GuanLuo) [inline docs/build.md:226]: `... you may need to disable...`

File: docs/build.md [hunk 2]
```diff
@@ -29,33 +29,68 @@
...
+
+* Fetch the appropriate \<xx.yy\>-py3-min image from NGC. The *min*
+  image is a minimal/base image that contains the CUDA, cuDNN,
+  TensorRT and other dependencies that are required to build Triton.
+
+* Create a *tritonserver_buildbase* Docker image that adds additional
+  build dependencies to the *min* image.
+
+* Run build.py within the *tritonserver_buildbase* image to actually
+  build Triton. See [Build without Docker](#ubuntu-without-docker) for
```
Comment (CoderHam) [inline docs/build.md:79]: Should we clarify that only Triton server and the backends are built by build.py (and not client libraries/ test utilities)?

File: docs/build.md [hunk 3]
```diff
@@ -69,62 +104,147 @@ build. For example, if you have a branch called "mybranch" in the
...
+master branch to build from the development branch). Then run build.py
+as described below.  The build.py script will perform the following
+steps (note that if you are building with Docker that these same steps
+will be performed during the Docker build within the
+*tritonserver_build* container).
+
+* Use the CMake files in [build](../build) to build Triton's core
+  shared library and *tritonserver* executable.
+
+* Fetch each request backend and build it using the CMake file from
```
Comment (CoderHam) [inline docs/build.md:134]: each **requested** backend

File: docs/build.md [hunk 4]
```diff
@@ -69,62 +104,147 @@ build. For example, if you have a branch called "mybranch" in the
...
+Triton. See the build.py invocation in [Build using
+Docker](#ubuntu-docker) for an example of how to run build.py. You can
+use that same invocation with the --no-container-build flag to build
+without Docker.
+
+The first step for any build is to checkout the
+[triton-inference-server/server](https://github.com/triton-inference-server/server)
+repo branch for the release you are interested in building (or the
+master branch to build from the development branch). Then run build.py
+as described below.  The build.py script will perform the following
```
Comment (CoderHam) [inline docs/build.md:126]: Remove extra space

File: docs/build.md [hunk 5]
```diff
@@ -69,62 +104,147 @@ build. For example, if you have a branch called "mybranch" in the
...
+
+* Fetch each repository agent and build it using the CMake file from
+  the corresponding repo. For example, the
+  [Checksum](https://github.com/triton-inference-server/checksum_repository_agent)
+  repository agent is built using
+  [triton-inference-server/checksum_repository_agent/CMakeLists.txt](https://github.com/triton-inference-server/checksum_repository_agent/blob/main/CMakeLists.txt).
+
 By default build.py clones Triton repos from
-https://github.com/triton-inference-server. Use the
+https://github.com/triton-inference-server.  Use the
```
Comment (CoderHam) [inline docs/build.md:153]: Remove extra space

File: docs/build.md [hunk 6]
```diff
@@ -69,62 +104,147 @@ build. For example, if you have a branch called "mybranch" in the
...
+
+*Under Construction*
+
+## Building on Unsupported Platforms
+
+Building for an unsupported OS and/or hardware platform is
+possible. All of the build scripting and CMake files are included in
+the public repos. However, due to differences in compilers, libraries,
+package management, etc. you may have to make changes in the build
+scripts and CMake files and even in the source code.
```
Comment (CoderHam) [inline docs/build.md:200]: build scripts, CMake files and even the source code

File: docs/build.md [hunk 7]
```diff
@@ -69,62 +104,147 @@ build. For example, if you have a branch called "mybranch" in the
...
+  for your platform you will need download the TensorFlow container
+  and modify its build to produce shared libraries for your
+  platform. You must use the TensorFlow source and build scripts from
+  within the NGC container because they contain Triton-specific
+  patches that are required for the Triton TensorFlow backend.
+
+* By default, the
+  [PyTorch](https://github.com/triton-inference-server/pytorch_backend)
+  backend build extracts pre-built shared libraries from an NGC
+  container. But the build can also use PyTorch shared libaries that
```
Comment (CoderHam) [inline docs/build.md:248]: shared **libraries** that

File: docs/build.md [hunk 8]
```diff
@@ -69,62 +104,147 @@ build. For example, if you have a branch called "mybranch" in the
...
+  Ubuntu-20.04 / x86-64 and so if you require the TensorFlow backend
+  for your platform you will need download the TensorFlow container
+  and modify its build to produce shared libraries for your
+  platform. You must use the TensorFlow source and build scripts from
+  within the NGC container because they contain Triton-specific
+  patches that are required for the Triton TensorFlow backend.
+
+* By default, the
+  [PyTorch](https://github.com/triton-inference-server/pytorch_backend)
+  backend build extracts pre-built shared libraries from an NGC
```
Comment (CoderHam) [inline docs/build.md:247]: from the PyTorch NGC container.

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: docs/build.md (hunk 2 above)
Comment (deadeyegoodwin) [reply docs/build.md:79]: I added a sentence at the beginning.

**Pull request merged**